### ISSUE TITLE
5.2.0 regression: tooltips never close when they are initialized using as `selector` option the attribute `title`

### ISSUE THREAD

**Issue by BrokenSourceCode** [twbs/bootstrap#36813]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

Hi,

Since Bootstrap v5.2.0, the tooltips never close when they are initialized with `selector` option. It worked fine under Bootstrap v5.1.3. So, I think there is a regression.

HTML:
```html
<div id="btn-box" class="btn-group m-5">
  <button class="btn btn-lg btn-primary" title="Label 1">Button 1</button>
  <button class="btn btn-lg btn-secondary" title="Label 2">Button 2</button>
</div>
```

JavaScript:
```js
new bootstrap.Tooltip('#btn-box', {
    selector: '[title]'
});
```

### Reduced test cases

| The tooltip is hidden when the mouse leaves the element | Version | `selector` option |                  CodePen link                   |
| :-----------------------------------------------------: | :-----: | :---------------: | :---------------------------------------------: |
|                           ✔️                            |  5.1.3  |        ❌         | https://codepen.io/BrokenSourceCode/pen/zYWzZNJ |
|                           ✔️                            |  5.1.3  |        ✔️         | https://codepen.io/BrokenSourceCode/pen/vYRZxNo |
|                           ✔️                            |  5.2.0  |        ❌         | https://codepen.io/BrokenSourceCode/pen/GRxEWWp |
|                           ❌                            |  5.2.0  |        ✔️         | https://codepen.io/BrokenSourceCode/pen/ExEXWPp |

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Chrome

### What version of Bootstrap are you using?

v5.2.0

**Comment by aavmurphy** [comment]
Same error for me.
Please could the documentation be updated for this feature - it's kind of hidden away, and the offsite links use jQuery and are older bootstrap versions. It's more than worthy enough to get its own example!

**Comment by julien-deramond** [comment]
_Apparently, it comes from https://github.com/twbs/bootstrap/commit/328f723008cc39292a9f355e2eafb0fd04740656._

I'll try an explanation and @GeoSot will confirm or not.

When the tooltip appears, `title` is removed from the DOM so the selector can't work anymore if the attribute is removed.

In your case, I'd say you can use `data-bs-title` instead of `title`.

```html
<div id="btn-box" class="btn-group m-5">
  <button class="btn btn-lg btn-primary" data-bs-title="Label 1">Button 1</button>
  <button class="btn btn-lg btn-secondary" data-bs-title="Label 2">Button 2</button>
</div>
```

```js
new bootstrap.Tooltip('#btn-box', {
  selector: '[data-bs-title]'
})
```

**Comment by GeoSot** [comment]
> Apparently, it comes from https://github.com/twbs/bootstrap/commit/328f723008cc39292a9f355e2eafb0fd04740656.

You are right on this. It was something we had discussed together too, as the title was just left empty.

@BrokenSourceCode I would prefer to not 'fix' this as it wasn't documented and of course is a bad practice to leave a blank attribute there. So you could, may use any other attr

**Comment by aavmurphy** [comment]
@GeoScot It is documented in https://getbootstrap.com/docs/5.2/components/tooltips/#options and really does need to be fixed as its a regression from several previous releases

**Comment by aavmurphy** [comment]
@BrokenSourceCode  Its in [https://getbootstrap.com/docs/5.2/components/tooltips](https://getbootstrap.com/docs/5.2/components/tooltips/#options),
scroll down to 'Options',
then the `selector` row in the table, which says
"If a selector is provided, tooltip objects will be delegated to the specified targets. In practice, this is used to also apply tooltips to dynamically added DOM elements (jQuery.on support). See [this issue](https://github.com/twbs/bootstrap/issues/4215) and [an informative example](https://codepen.io/Johann-S/pen/djJYPb)."
The example uses jQuery, and a legacy version of bootstrap, so the documentation could be updated as this functionality makes tooltips *much* easier to use

**Comment by saturdaywalkers** [comment]
Not fixed in 5.2.1

**Comment by saturdaywalkers** [comment]
Thanks @BrokenSourceCode !

Turned out quite easy to fix... just grep the js html templates for `title=` and add a target to use in a selector, `data-bs-tooltip` in my case.
